packages/ra-core/src/form/Form.tsx

```diff
@@ -1,11 +1,15 @@
 import * as React from 'react';
-import { ReactNode } from 'react';
+import { ReactNode, useContext } from 'react';
 import {
     FormProvider,
     FieldValues,
     UseFormProps,
     SubmitHandler,
 } from 'react-hook-form';
+import {
+    UNSAFE_DataRouterContext,
```

Comment:
Unfortunately react-router doesn't expose the DataRouterContext, see https://github.com/remix-run/react-router/discussions/11347